bake/hclparser/hclparser.go

```diff
@@ -49,29 +51,38 @@ type parser struct {
 	attrs map[string]*hcl.Attribute
 	funcs map[string]*functionDef
 
-	blocks      map[string]map[string][]*hcl.Block
-	blockValues map[*hcl.Block]reflect.Value
-	blockTypes  map[string]reflect.Type
+	blocks       map[string]map[string][]*hcl.Block
+	blockValues  map[*hcl.Block][]reflect.Value
+	blockEvalCtx map[*hcl.Block][]*hcl.EvalContext
+	blockNames   map[*hcl.Block][]string
+	blockTypes   map[string]reflect.Type
 
 	ectx *hcl.EvalContext
 
-	progress  map[string]struct{}
-	progressF map[string]struct{}
-	progressB map[*hcl.Block]map[string]struct{}
-	doneF     map[string]struct{}
-	doneB     map[*hcl.Block]map[string]struct{}
+	progressV map[uint64]struct{}
+	progressF map[uint64]struct{}
+	progressB map[uint64]map[string]struct{}
+	doneB     map[uint64]map[string]struct{}
+}
+
+type internalGetEvalContexts interface {
```

Comment:
This should be public if we expect caller to implement it. And type checked with `var _ WithEvalContexts = &Group{}`

Reply:
Done. I've type checked for Group + Target for both the interfaces.